client/src/Nav.jsx

```diff
@@ -1,58 +1,74 @@
-import { AppBar, Typography, Toolbar, Button } from '@mui/material';
-import { useEffect, useState } from 'react';
-import { useNavigate } from 'react-router-dom';
+import { AppBar, Typography, Toolbar, Button } from "@mui/material";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
 
 const Nav = () => {
-
-  const [destination, setDestination] = useState('/');
+  const [destination, setDestination] = useState("/");
+  const [clicked, setClicked] = useState(false);
   const navigate = useNavigate();
 
   const handleClick = (e) => {
+    document.getElementById("nav").addEventListener("click", () => {
+      setClicked(true);
+    });
+
     const elText = e.target.innerText;
     switch (elText) {
-      case 'Dashboard':
-        setDestination('/dashboard'); // user dashboard page
+      case "Dashboard":
+        setDestination("/dashboard"); // user dashboard page
         break;
-      case 'Create':
-        setDestination('/') // No route yet
+      case "Create":
+        setDestination("/"); // No route yet
         break;
-      case 'SIGN UP':
-        setDestination('/register'); // register page
+      case "SIGN UP":
+        setDestination("/register"); // register page
         break;
-      case 'SIGN IN':
-        setDestination('/login'); // login page
+      case "SIGN IN":
+        setDestination("/login"); // login page
         break;
       default:
-        setDestination('/'); // Home/landing page
+        setDestination("/"); // Home/landing page
         break;
     }
   };
 
   useEffect(() => {
-    navigate(destination);
-  // eslint-disable-next-line react-hooks/exhaustive-deps
+    // Only redirect if the Nav Bar was clicked (not on page refresh)
```

Comment:
nice catch

Reply:
Removing Nav workaround entirely, Thanh patched in a separate PR